### ISSUE TITLE
Error reading NIFTI file

### ISSUE THREAD

**Issue by chrisgorgo** [bids-standard/legacy-validator#44]
Im getting this error when validating this file  (on the command line):

```
/Users/filo/drive/workspace/bids-validator/node_modules/nifti-js/nifti.js:25
    throw new Error("The buffer is not even large enough to contain the minimal header I would expect from a NIfTI file!")
    ^

Error: The buffer is not even large enough to contain the minimal header I would expect from a NIfTI file!
    at Object.parseNIfTIHeader (/Users/filo/drive/workspace/bids-validator/node_modules/nifti-js/nifti.js:25:11)
    at /Users/filo/drive/workspace/bids-validator/utils/files.js:125:36
    at Gunzip.onEnd (zlib.js:227:5)
    at emitNone (events.js:72:20)
    at Gunzip.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:903:12)
    at doNTCallback2 (node.js:439:9)
    at process._tickCallback (node.js:353:17)
orange:bids-validator filo$ open /tmp/AA//sub-S2529LVY1263171/func/sub-S2529LVY1263171_task-rest_acq-RL_run-2_sbref.nii.gz
```

**Comment by constellates** [comment]
The file you sent me is failing because it's not gzipped. Do we need to account for reading unzipped niftis and did we lose our check for niftis without the .gz extension? I'm not seeing it anymore.

**Comment by constellates** [comment]
Ignore that I believe I unzipped it by opening it from my downloads.

**Comment by constellates** [comment]
I haven't been able to reproduce this one. I was able to get it to validate by adding it to a dataset and testing w/ the CLI. I also created a quick script to log the contents. If you want to test it you can save it in the root of the validator repo and run it like `node nameOfFile.js /path/to/single/scan/file.nii.gz`

```
var files = require('./utils/files');
var file = {path: process.argv[2]};

files.readNiftiHeader(file, function (header) {
    console.log(header);
});
```

Is it possible your validator or node modules are out of date?

**Comment by chrisgorgo** [comment]
This is weird - I could not replicate the error with the code you provided running on individual nifti files. However when running on the whole dataset it still appears. I created a simple stripped down dataset to replicate the bug: https://docs.google.com/uc?id=0B77zr9yIiKOTRnZzVDN4akQweXM&export=download

**Comment by constellates** [comment]
Thanks. I'll test this out.

**Comment by constellates** [comment]
When I pulled that same file out of the dataset linked above I was able to reproduce the error with even the minimal test. It looks like that particular file shrunk during decompression.  

My header reading strategy was to read the number of bytes equal to the nifti header size (348) then decompress it and run the binary reading. I was assuming that headers would generally grow in size during decompression. This particular file shrunk so I increased the overall number of bytes we're reading for each scan from 348 to 500. I'm also looking to see if there are strategies for predicting decompression sizes or at least establishing a sane limit of bytes we need to read.
